edgelet/iotedged/src/lib.rs

```diff
@@ -538,7 +547,17 @@ where
     )
     .with_issuer(CertificateIssuer::DeviceCa);
 
-    let cert_manager = Arc::new(CertificateManager::new(crypto.clone(), edgelet_cert_props));
+    let cert_manager = CertificateManager::new(crypto.clone(), edgelet_cert_props).context(
+        ErrorKind::Initialize(InitializeErrorReason::CreateCertificateManager),
+    )?;
+
+    // Create the certificate management timer and channel
+    let (restart_tx, restart_rx) = oneshot::channel();
+    let expiration_timer = cert_manager
+        .schedule_expiration_timer(move || restart_tx.send(()).unwrap_or(()))
```

Comment:
If `send` fails then that means the receiver has already been de-allocated. Should we be ignoring that case?

Reply:
Good question - I was following the pattern that we were doing with OneShot channels, but added error handling since that's the correct thing to do :)